### PR TITLE
Add proto for MeshCA MeshCertificateService

### PR DESCRIPTION
**Pull request by yashykt**
Add proto for MeshCA MeshCertificateService

Looking for review from @donnadionne 
cc @markdroth

### CODE REVIEW

File: src/proto/grpc/tls/provider/meshca/meshca.proto [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package google.security.meshca.v1;
```
Comment (markdroth) [inline src/proto/grpc/tls/provider/meshca/meshca.proto:17]: I think that for now, we should be using the copy of this from the grpc-proto repo:

https://github.com/grpc/grpc-proto/blob/master/grpc/tls/provider/meshca/experimental/config.proto

Comment (sanjaypujare) [reply src/proto/grpc/tls/provider/meshca/meshca.proto:17]: That is different: that's the meshCA config defined as proto (more relevant when it comes from the control plane). Yash needs the proto for MeshCA request/response/service which is not there in grpc-proto AFAIK.

Comment (yashykt) [reply src/proto/grpc/tls/provider/meshca/meshca.proto:17]: Right, Mark, we are talking about different protos. The one that we need here is the proto for the MeshCA service.

Comment (markdroth) [reply src/proto/grpc/tls/provider/meshca/meshca.proto:17]: Doh!  You're right.  I wasn't reading closely.  Sorry for the confusion.

Comment (yashykt) [reply src/proto/grpc/tls/provider/meshca/meshca.proto:17]: For the one that you are referring to is the config.proto and for that I'll probably take the approach that we are already following everywhere else for consistency.

**Comment by yashykt** [comment]
I thought about that, but adding an entire submodule for a single proto seems to be an overkill. If we end up needing more stuff, we can probably make it a submodule (though it'll be pretty wild if istio ever ends up deciding to use grpc as a submodule, i wonder how it would work then)

**Comment by markdroth** [comment]
Okay.  How about at least putting this file in the `third_party` tree?  That way, it's clear that it's not ours, and it will be automatically exempt from things like the include check and copyright check.

**Comment by yashykt** [comment]
I am fine putting it in the third_party directory but it might cause issues with our build and import scripts. I'd like Nico's blessing on that. @nicolasnoble

**Comment by yashykt** [comment]
I think I'd like to get unblocked on this. But, if we decide to move it to third_party we should probably do it for all our third_party protos including those in src/proto/grpc/testing/xds.

**Comment by markdroth** [comment]
I think we should move this proto to third_party.  This is the standard way we do things like this; in general, the only protos in src/proto are the ones our team owns or co-owns.

Unfortunately, the protos in src/proto/grpc/testing/xds are an exception to this, and they're a horrible hack.  Those protos already exist in third_party/envoy-api, and we use the copies there to generate the upb code that we use inside of core.  But xds_end2end_test uses the protobuf library, and it can't directly depend on the protos from third_party/envoy-api due to a horrible circular dependency problem in bazel: the build rules from the envoy repo depend on the build rules in the grpc repo, because they use the grpc_cc_library rule that we provide.  So as a horrible hack, we copied the protos we need for the test into our own location, and we use them from there in xds_end2end_test only.  They should not be used anywhere else, but we can't easily get rid of them.

**Comment by yashykt** [comment]
> I think we should move this proto to third_party. This is the standard way we do things like this; in general, the only protos in src/proto are the ones our team owns or co-owns.
That is not true. Even if we ignore the xds protos, we still have tools/grpcz/census.proto and src/proto/grpc/status/status.proto

Additionally, the files in third_party/... normally also have a counterpart in internal third_party/... and hence we do not need to copy any of the files from our repo's third_party internally. You would notice that internally we do not have a third_party/grpc/third_party. Adding this proto file to third_party and making changes to our import scripts is possible but that would be a first for our repo.

**Comment by yashykt** [comment]
I had a chat with Nico on how we want to do this. The idea was that we SHOULD place it in third_party/ and try to keep a single source of truth for the proto internally.

**Comment by yashykt** [comment]
The issue on how to manage the dependency when we stop generating upb files still exists though, but that seems solvable.

**Comment by yashykt** [comment]
Issues : #23028

**Comment by yashykt** [comment]
ick, turns out that the internal proto that I was referring to was actually being imported by grpc-java. Istio is not being imported in google3 right now.

After chatting with @nicolasnoble again, the decision was to move it back to src/proto till atleast a time when there is a reliable internal third_party/ source that we can depend on.

**Pull request merged**